include/swift/AST/Decl.h

```diff
@@ -3778,10 +3777,17 @@ class ClassDecl final : public NominalTypeDecl {
     /// The superclass type and a bit to indicate whether the
     /// superclass was computed yet or not.
     llvm::PointerIntPair<Type, 1, bool> SuperclassType;
+
+    /// Whether this class has any designated initializers that subclasses cannot access.
+    unsigned HasMissingDesignatedInitializers : 1;
+
+    /// Whether we have computed the above field or not.
+    unsigned HasMissingDesignatedInitializersComputed : 1;
```

Comment:
I think it's fine to not have a cached bit for this at all. Just checking the attribute is fine.